### DISCUSSION TITLE
How to set the registry for the base image

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/sdk-container-builds#580] [Ideas]
Sdk builds infer the base image using TargetFramework among other things, however if you want to use a hosted registry you would have to set the fully qualified image in `ContainerBaseImage` which overrides the project-dependent logic used to choose the base image.

> If you set a value here, you should set the fully-qualified name of the image to use as the base, including any tag you prefer

Using an option like `ContainerBaseImageRegistry` could only hint the registry and still hand off the actual image selection to the sdk.

**Comment by baronfel** [comment]
How would we have any idea if the registry even had the images that we expect? The only reasonable defaults that we have any knowledge of are the Microsoft images. 

If an organization wanted to do similar inference then I'd suggest that they create a nuget package with MSBuild logic to do that inference. That allows them to encode whatever patterns they like, and doesn't lead to the SDK making incorrect inference.

**Reply by alrz** [reply]
> How would we have any idea if the registry even had the images that we expect?

I should've mentioned that the registry would be a cached proxy to the original mcr so all images would be available
